### PR TITLE
Consolidate AWS credentials, Cloudwatch dimension wilcards

### PR DESCRIPTION
**Pull request by johnrengelman**
Rebased & retargeted version of https://github.com/influxdata/telegraf/pull/1155

### CODE REVIEW

**Comment by sparrc** [comment]
few things:

1. update CHANGELOG.md
1. There are some READMEs that will need to be updated for each aws plugin

**Comment by johnrengelman** [comment]
I've updated the ChangeLog and the READMEs

**Comment by sparrc** [comment]
LGTM but please rebase your changes so I can merge

**Comment by johnrengelman** [comment]
@sparrc rebased and I relocate my changelog entries to to "unreleased" section.

**Comment by johnrengelman** [comment]
And rebased again.
Having the changelog submitted as part of the PR is quite a pain. it means that every PR has to be rebased immediately before merging because they all change the same file.

**Comment by sparrc** [comment]
sorry but it's not always this painful, you've made quite a large change to the changelog at the same time that I was making a release....

**Comment by sparrc** [comment]
thanks @johnrengelman

File: plugins/inputs/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -106,20 +121,40 @@ func (c *CloudWatch) Gather(acc telegraf.Accumulator) error {
...
 		for _, m := range c.Metrics {
-			dimensions := make([]*cloudwatch.Dimension, len(m.Dimensions))
-			for k, d := range m.Dimensions {
-				dimensions[k] = &cloudwatch.Dimension{
-					Name:  aws.String(d.Name),
-					Value: aws.String(d.Value),
+			if !hasWilcard(m.Dimensions) {
+				dimensions := make([]*cloudwatch.Dimension, len(m.Dimensions))
+				for k, d := range m.Dimensions {
+					fmt.Printf("Dimension [%s]:[%s]\n", d.Name, d.Value)
```
Comment (cbandy) [inline plugins/inputs/cloudwatch/cloudwatch.go]: Is this print intentional?

Comment (johnrengelman) [inline plugins/inputs/cloudwatch/cloudwatch.go]: Nope, that's was some lingering debugging. It should get removed.

**Pull request merged**